mithril-aggregator/src/artifact_builder/cardano_database_artifacts/ancillary.rs

```diff
@@ -107,4 +232,100 @@ mod tests {
             ]
         );
     }
+
+    #[tokio::test]
+    async fn create_archive_should_embed_ledger_volatile_directories_and_last_immutables() {
+        let test_dir = "cardano_database/create_archive";
+        let cardano_db = DummyCardanoDbBuilder::new(test_dir)
+            .with_immutables(&[1, 2, 3])
+            .with_ledger_files(&["blocks-0.dat", "blocks-1.dat", "blocks-2.dat"])
+            .with_volatile_files(&["437", "537", "637", "737"])
+            .build();
+        std::fs::create_dir(cardano_db.get_dir().join("whatever")).unwrap();
+
+        let db_directory = cardano_db.get_dir().to_path_buf();
+        let snapshotter = {
+            CompressedArchiveSnapshotter::new(
+                db_directory.clone(),
+                db_directory.parent().unwrap().join("snapshot_dest"),
+                SnapshotterCompressionAlgorithm::Gzip,
+                TestLogger::stdout(),
+            )
+            .unwrap()
+        };
+
+        let builder = AncillaryArtifactBuilder::new(
+            vec![],
+            Arc::new(snapshotter),
+            CardanoNetwork::DevNet(123),
+            CompressionAlgorithm::Gzip,
+            TestLogger::stdout(),
+        );
+
+        let snapshot = builder
+            .create_ancillary_archive(&CardanoDbBeacon::new(99, 1))
```

Comment:
This would be closer to reality:
```suggestion
            .create_ancillary_archive(&CardanoDbBeacon::new(99, 2))
```

Reply:
Done ✅